jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ResourceSelector.kt

```diff
@@ -205,7 +198,35 @@ class ResourceSelector<T> @JvmOverloads constructor(
         LOADED
     }
 
-    private companion object {
-        val LOG = getLogger<ResourceSelector<*>>()
+    companion object {
+        private val LOG = getLogger<ResourceSelector<*>>()
+        @JvmStatic
+        fun builder(project: Project) = object : ResourceBuilder {
+            override fun <T> resource(resource: () -> Resource<out Collection<T>>): OtherOptions<T> = OtherOptions(project, resource)
+        }
+    }
+
+    interface ResourceBuilder {
+        fun <T> resource(resource: Resource<out Collection<T>>): OtherOptions<T> = resource { resource }
+        fun <T> resource(resource: () -> Resource<out Collection<T>>): OtherOptions<T>
+    }
+
+    class OtherOptions<T> internal constructor(private val project: Project, private val resource: () -> Resource<out Collection<T>>) {
```

Comment:
Hmmm... Maybe not an issue, but seems weird to call this OtherOptions since that is what the IDE will say the type is.